Create a PR for PM Tien Tran to review

ttuan-evizi opened the pull request.

ttuan-evizi committed:
push code

ttuan-evizi committed:
Delete unnecessary files, group files into folders, remove unnecessary comments

ttuan-evizi committed:
Update README.md

ttuan-evizi committed:
Change files' structure, update README

ttuan-evizi committed:
Add another example of OOP using Encapsulation

ttuan-evizi committed:
add new functions to main. and update README

kahnext commented on src_Car/SuperCar.ts:
```diff
@@ -0,0 +1,37 @@
+import { Vehicle } from "./Vehicle";
```
do not create bunch of `src_` folder, let's organize it with sub folder instead

ttuan-evizi replied:
yes, ill combine them into 1 folder src, with subfolders such as Car, and Geometry

kahnext commented on src_Car/SuperCar.ts:
```diff
@@ -0,0 +1,37 @@
+    }
+
+    // Public method to get top speed
+    public getTopSpeed(): number {
```
don't see any override and overload in this inheritance Vehicle <> SuperCar

ttuan-evizi replied:
added overloading to SuperCar

kahnext commented on src_Geometry/Triangle.ts:
```diff
@@ -0,0 +1,16 @@
+import { Shape } from './Shape';
+
+export class Triangle extends Shape {
```
still don't see any overloading support within this Geometry implementation

ttuan-evizi replied:
added overloading to Triangle, Circle, and other shapes as well

ttuan-evizi committed:
Change file structure according to the critics

ttuan-evizi committed:
add overloading to SuperCar.ts

ttuan-evizi committed:
add overloading to geometry and update README

ttuan-evizi committed:
add a new SuperCar object to test out a function

ttuan-evizi committed:
adding super to child's method to implement overriding

ttuan-evizi committed:
Create Phone

ttuan-evizi committed:
create another example of OOP

ttuan-evizi committed:
add files skeleton, and code setup the Phone class

ttuan-evizi committed:
update Samsung.ts, and iPhone.ts

- create classes, and implement abstract methods
- Attempt to overload & overriding methods

ttuan-evizi committed:
Update SuperCar.ts

kahnext commented on src/Geometry/Rectangle.ts:
```diff
@@ -0,0 +1,18 @@
+        super();
+    }    
+    // Overloading
+    calculateArea(): number;
```
it's not overload anything. I don't see several methods with same name here.

ttuan-evizi replied:
I have added implementations to these functions.

kahnext commented on src/Geometry/Triangle.ts:
```diff
@@ -0,0 +1,21 @@
+
+    // Overloading
+    calculateArea(): number;
+    calculateArea(base: number, height: number): number;
```
where is implementation here? It's just a declaration.

ttuan-evizi replied:
I have added implementations to these functions.

kahnext commented on src/Geometry/main.ts:
```diff
@@ -0,0 +1,27 @@
+import { Square } from './Square';
+import { Triangle } from './Triangle';
+import { Trapezoid } from './Trapezoid';
+function main() {
```
do not spam bunch of main file. How if we just create relevant method with specific name like `geo_test()`, `car_test()`, etc..
Then having only one main file includes above functions.

ttuan-evizi replied:
I have moved main to root, and created relevant method with specific name like geo_test(), car_test() that will be run using main.

kahnext commented on src/Phone/iPhone.ts:
```diff
@@ -0,0 +1,25 @@
+
+    public call(): string;
+    public call(phoneNumber: string): string;
+    public call(phoneNumber?: string) {
```
good overloading support

kahnext commented on src/Phone/SamsungS24.ts:
why do we push empty file.

ttuan-evizi replied:
They have been removed. I should have checked before pushing them. My bad.

kahnext commented on src/Phone/iPhone16.ts:
why do we push empty file

ttuan-evizi replied:
They have been removed. I should have checked before pushing them. My bad.

kahnext commented on src/Phone/main.ts:
it's empty file, please remove.

ttuan-evizi replied:
They have been removed. I should have checked before pushing them. My bad.

ttuan-evizi committed:
Remove empty files, move and combine main

Remove empty .ts files:
- iPhone16.ts
- SamsungS24.ts

Combine different mains  in src into one main and move it to root.

ttuan-evizi committed:
Update main, and implement overloading in Geometry, and remove unuse Overloading

kahnext reviewed (approved):
LGTM
